agents/wiregrid_actuator/wiregrid_actuator.py

```diff
@@ -0,0 +1,649 @@
+                return False, 'Could not acquire lock in eject().'
+
+            if self.controlling :
+                self.log.warn('Actuator is controlled by another function.')
+                return False, 'Actuator control is held by another function.'
+         
+            # Running
+            self.controlling = True
+            time.sleep(1)
+         
+            # Moving commands
+            ret = self.__eject(850, 1.0)
+            if not ret :
+                self.log.error('Failed to insert!')
+                return False, 'Failed eject() in __eject(850,1.0)'
+         
+            # Finishing
+            self.controlling = False
+            pass
+
+        return True, 'Finish eject()'
+
+
+    def insert_homing(self, session, params=None):
+        with self.lock.acquire_timeout(timeout=3, job='insert_homing') as acquired:
+            if not acquired:
+                self.log.warn('Lock could not be acquired because it is held by {}.'.format(self.lock.job))
+                return False, 'Could not acquire lock in insert_homing().'
+
+            if self.controlling :
+                self.log.warn('Actuator is controlled by another function.')
+                return False, 'Actuator control is held by another function.'
+         
+            # Running
+            self.controlling = True
+            time.sleep(1)
+         
+            # Moving commands
+            ret = self.__insert(1000, 0.1)
+            if not ret :
+                self.log.error('Failed to insert_homing!')
+                return False, 'Failed insert_homing() in __insert(1000,0.1)'
+         
+            # Finishing
+            self.controlling = False
+            pass
+
+        return True, 'Finish insert_homing()'
+
+
+    def eject_homing(self, session, params=None):
+        with self.lock.acquire_timeout(timeout=3, job='eject_homing') as acquired:
+            if not acquired:
+                self.log.warn('Lock could not be acquired because it is held by {}.'.format(self.lock.job))
+                return False, 'Could not acquire lock in eject_homing().'
+
+            if self.controlling :
+                self.log.warn('Actuator is controlled by another function.')
+                return False, 'Actuator control is held by another function.'
+         
+            # Running
+            self.controlling = True
+            time.sleep(1)
+         
+            # Moving commands
+            ret = self.__eject(1000, 0.1)
+            if not ret :
+                self.log.error('Failed to eject_homing!')
+                return False, 'Failed eject_homing() in __eject(1000,0.1)'
+         
+            # Finishing
+            self.controlling = False
+            pass
+
+        return True, 'Finish eject_homing()'
+
+
+    def stop(self, session, params=None):
+        self.log.warn('Try to stop and hold the actuator.')
```

Comment:
This stop command is not used in a normal operation.
This command is assumed to be used when you want to stop the movement of the actuator in an emergency.
So, I assigned all the logs to warn or error.

Reply:
Gotcha, I guess I'm thinking more in the programmatic sense that nothing here as actually gone wrong within the code. I don't feel strongly about needing it to be in `info` though, so that's fine.